src/core.rs

```diff
@@ -663,12 +663,18 @@ impl Core {
 
     /// Checks if the given name is missing from our routing table. If so, tries to refill the
     /// bucket.
-    fn harvest_node(&mut self, name: &XorName) -> Result<(), RoutingError> {
-        if self.connection_filter.insert(name) == 0 && self.routing_table.need_to_add(name) {
-            let i = self.name().bucket_index(name);
-            self.request_bucket_ids(i)
-        } else {
-            Ok(())
+    fn harvest_node(&mut self, name: &XorName, routing_msg: &RoutingMessage) -> Result<(), RoutingError> {
+        match routing_msg {
+            &RoutingMessage::Response(ResponseMessage { content: ResponseContent::GetSuccess(..), .. }) => {
+                if self.connection_filter.insert(name) == 0 && self.routing_table.need_to_add(name) {
```

Comment:
I'd insert `i` into a `bucket_filter` instead of inserting `name` into the `connection_filter` (I think this is the only place we use that filter, right?): If we discover two nodes from the same bucket, it suffices to send one `GetCloseGroup` request.